components/api/db/cellery_hub_db.bal

```diff
@@ -537,6 +496,48 @@ returns json | error {
     return check json.convert(imagesListResponse);
 }
 
+public function getArtifactListLength(string imageId, string artifactVersion) returns int | error {
+    log:printDebug(io:sprintf("Retriving artifact count for image ID : %s and image version %s", imageId, artifactVersion));
+    table<gen:Count> res = check connection->select(GET_ARTIFACT_COUNT, gen:Count, imageId, artifactVersion, loadToMemory = true);
+    int length = check getTotalRecordsCount(res);
+    return length;
+}
+
+function buildJsonPayloadForGetArtifact(table< record {}> res, string orgName, string imageName, string artifactVersion) returns json | error {
+    if (res.count() == 1) {
+        json resPayload = {};
+        gen:ArtifactResponse artRes = check gen:ArtifactResponse.convert(res.getNext());
+        string metadataString = encoding:byteArrayToString(artRes.metadata);
+        io:StringReader sr = new(metadataString, encoding = "UTF-8");
+        json metadataJson = check sr.readJson();
+        resPayload.description = encoding:byteArrayToString(artRes.description);
+        resPayload.pullCount = artRes.pullCount;
+        resPayload.lastAuthor = artRes.lastAuthor;
+        resPayload.updatedTimestamp = artRes.updatedTimestamp;
+        resPayload.metadata = metadataJson;
+        return resPayload;
+    } else if (res.count() == 0) {
+        log:printDebug(io:sprintf("The requested artifact \'%s/%s:%s\' was not found in REGISTRY_ORGANIZATION",
```

Comment:
Better to not log this at this level. (If a log is required at this level, the log message should be changed.)